src/Contracts/PaginatesEloquentModels.php

```diff
@@ -11,18 +11,20 @@ interface PaginatesEloquentModels
      *
      * @param  \Laravel\Scout\Builder  $builder
      * @param  int  $perPage
+     * @param  string  $pageName
      * @param  int  $page
      * @return \Illuminate\Contracts\Pagination\LengthAwarePaginator
      */
-    public function paginate(Builder $builder, $perPage, $page);
+    public function paginate(Builder $builder, $perPage, $pageName, $page);
```

Comment:
Wondering if we can instead do the below to make it less breaking?

```suggestion
    public function paginate(Builder $builder, $perPage, $page, $pageName = 'page');
```

Reply:
Yeah, I did consider this option. However, being that it's an interface, all implementations will still break won't they? And by adding to the end, we break uniform with how all the other `paginate` methods are defined 🤔

Reply:
However, happy to change to this if we think that's the best option; just providing my reasoning. LMK 👍

Reply:
It's not so much about the implementation but about usage. End users who use scout and not implement their own engine will have a smooth upgrade path. Otherwise you still need to move the "page" argument to the fourth one.

Reply:
Don't the end users use the `Builder` rather than the engines directly? So their usage doesn't change.

Reply:
Yeah that's true. Let's see what @taylorotwell says.